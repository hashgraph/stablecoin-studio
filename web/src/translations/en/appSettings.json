{
	"title": "Configure custom RPC/MIRROR",
	"subtitle": "Choose an option",
	"name": "Name",
	"rpc": "RPC",
	"mirrorNode": "Mirror node",
	"url": "Url",
	"apiKey": "Api Key",
	"addMirror": "Add new service",
	"addRPC": "Add RPC",
	"removeMirror": "Remove Mirror Node",
	"removeRpc": "Remove RPC",
	"header": "Header",
	"mirrorNameExists": "Mirror name already exists, if you want to update it remove it first and recreate it",
	"rpcNameExists": "RPC name already exists, if you want to update it remove it first and recreate it"
}
