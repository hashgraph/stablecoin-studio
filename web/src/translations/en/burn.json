{
    "title": "Burn tokens operation",
    "operationTitle": "Operation details",
    "amountLabel": "Enter the amount",
    "amountPlaceholder": "0",
    "originAccountPlaceholder": "0.0.0",
    "modalSuccessDesc": "{{amount}} tokens have been burned.",
    "modalAction": {
        "subtitle": "Operation details review",
        "originAccount": "Account",
        "amount": "Amount",
        "accept": "Burn tokens"
    }
}