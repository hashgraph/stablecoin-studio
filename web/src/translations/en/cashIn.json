{
    "title": "Cash in operation",
    "operationTitle": "Operation details",
    "amountLabel": "Introduce amount",
    "amountPlaceholder": "0",
    "destinationAccountLabel": "Destination account",
    "destinationAccountPlaceholder": "0.0.0",
    "modalSuccessDesc": "The cash in operation finished successfully.",
    "modalAction": {
        "subtitle": "Operation details review",
        "destinationAccount": "Destination account",
        "amount": "Amount",
        "accept": "Cash in tokens"
    }
}