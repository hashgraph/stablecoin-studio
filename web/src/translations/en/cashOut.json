{
    "title": "Cash out operation",
    "operationTitle": "Operation details",
    "amountLabel": "How much tokens?",
    "amountPlaceholder": "Introduce amount",
    "originAccountLabel": "From which account tokens will be withdrawn?",
    "originAccountPlaceholder": "Introduce account",
    "modalSuccessDesc": "The cash out operation finished successfully.",
    "modalAction": {
        "subtitle": "Operation details review",
        "originAccount": "Account to cash out",
        "amount": "Amount",
        "accept": "Make cash out"
    }
}