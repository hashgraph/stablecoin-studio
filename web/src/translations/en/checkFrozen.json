{
	"title": "Check account frozen status",
	"operationTitle": "Operation details",
	"accountLabel": "Enter the account",
	"accountPlaceholder": "0.0.0",
	"modalIsFrozen": "The account {{ account }} is frozen for the token.",
	"modalIsNotFrozen": "The account {{ account }} is unfrozen for the token.",
	"modalAction": {
		"subtitle": "Operation details review",
		"account": "Account",
		"accept": "Get account frozen status"
	}
}
