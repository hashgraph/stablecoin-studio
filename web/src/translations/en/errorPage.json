{
  "stableCoinNotSelected": {
    "title": "Choose a stable coin to start operating!",
    "description": "You have not chosen a stable coin yet.",
    "description2": "Choose an item to operate."
  }
}