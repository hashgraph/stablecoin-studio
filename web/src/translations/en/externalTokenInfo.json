{
    "common": {
      "nextStep" : "Next step",
      "goBack" : "Go back",
      "createStableCoin": "Create stable coin",
      "createNewStableCoin": "Import token",
      "cancel": "Cancel",
      "addExternalToken": "Import token"
    },
    "tabs": {
      "externalTokenInfo": "Token info"
    },
    "externalTokenInfo": {
      "stableCoinId": "Stable coin id",
      "stableCoinIdPlaceholder": "0.0.0",
      "title": "Imported token",
      "autoCheckRoles": "Do you want to get the roles automatically? This operation will incur a cost",
      "rolesPlaceholder": "Select the roles"
    },
    "optionalDetails": {
      "decimals": "Decimals",
      "expirationDate": "Expiration date",
      "finite": "Finite",
      "initialSupply": "Initial supply",
      "infinite": "Infinite",
      "placeholder": "Enter the {{placeholder}}",
      "maxSupply": "Max supply",
      "title": "Stable coin optional details",
      "typeSupply": "Type supply",
      "typeSupplyPlaceholder": "Choose type supply"
    },
    "managementPermissions": {
      "currentUserKey": "Current user key",
      "description": "Allow all the following features to be managed through smart contracts",
      "pause": "Pause",
      "freeze": "Freeze",
      "introduce": "Enter the {{name}}",
      "none": "None",
      "otherKey": "Other key",
      "placeholder": "Enter the {{placeholder}}",
      "rescue": "Rescue",
      "admin": "Admin",
      "supply": "Supply",
      "theSmartContract": "The smart contract",
      "title": "Stable coin management permissions",
      "totalSupply": "Total supply",
      "treasuryAccountAddress": "Treasury account address",
      "wipe": "Wipe"
    },
    "review": {
      "basicDetails": "Basic details",
      "managementPermissions": "Management permissions",
      "optionalDetails": "Optional details",
      "title": "Stable coin review"
    },
    "notification": {
      "title": "{{ result }} creating the stable coin",
      "descriptionSuccess": "The stable coin was created",
      "descriptionError": "Sorry, there was an error creating the stable coin"
    }
  }