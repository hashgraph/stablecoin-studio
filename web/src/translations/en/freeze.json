{
    "title": "Freeze operation",
    "operationTitle": "Operation details",
    "accountLabel": "Enter the account",
    "accountPlaceholder": "0.0.0",
    "modalSuccess": "The account {{ account }} is frozen.",
    "modalAction": {
        "subtitle": "Operation details review",
        "account": "Account",
        "accept": "Freeze account"
    }
}