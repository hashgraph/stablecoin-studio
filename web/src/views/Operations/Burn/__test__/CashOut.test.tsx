import userEvent from '@testing-library/user-event';
import { waitFor } from '@testing-library/react';
import { render } from '../../../../test/index';
import en from '../../../../translations/en/burn.json';
import BurnOperation from '..';

const translations = en;

describe(`<${BurnOperation.name} />`, () => {
	test('should render correctly', () => {
		const component = render(<BurnOperation />);

		expect(component.asFragment()).toMatchSnapshot();
	});

	test('should render titles', () => {
		const component = render(<BurnOperation />);

		expect(component.getByTestId('title')).toHaveTextContent(translations.title);
		expect(component.getByTestId('operation-title')).toHaveTextContent(translations.operationTitle);
	});

	test('should have an input to write the amount', () => {
		const component = render(<BurnOperation />);

		expect(component.getByTestId('amount')).toBeInTheDocument();
	});

	test('should have a disabled confirm button that is enable when introduce valid data', async () => {
		const component = render(<BurnOperation />);

		const button = component.getByTestId('confirm-btn');
		expect(button).toBeDisabled();

		const amount = component.getByTestId('amount');
		userEvent.type(amount, '10000');

		await waitFor(() => {
			expect(button).toBeEnabled();
		});
	});
});
