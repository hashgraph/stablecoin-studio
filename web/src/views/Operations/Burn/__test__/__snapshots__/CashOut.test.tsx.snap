// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BurnOperation /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-16oka5y"
  >
    <h2
      class="chakra-heading css-llrpa2"
      data-testid="base-container-heading"
    >
      Operations
    </h2>
    <div
      class="css-1b6lda7"
      data-testid="base-container-children"
    >
      <div
        class="css-1us6gcn"
      >
        <div
          class="css-1r9sqqd"
        >
          <div
            class="chakra-stack css-n21gh5"
          >
            <h2
              class="chakra-heading css-17ipgvs"
              data-testid="title"
            >
              Burn tokens operation
            </h2>
            <p
              class="chakra-text css-1lyqlf8"
              data-testid="operation-title"
            >
              Operation details
            </p>
            <form
              class="chakra-stack css-b5leqj"
            >
              <div
                class="chakra-stack css-1h0sumw"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  data-testid="formControl"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-1hfoxt1"
                    for="field-:r1:"
                    id="field-:r1:-label"
                  >
                    <div
                      class="chakra-stack css-84zodg"
                    >
                      <p
                        class="chakra-text css-mgupmj"
                      >
                        How much tokens?
                      </p>
                      <p
                        class="chakra-text css-ftb7fv"
                        data-testid="required"
                      >
                        *
                      </p>
                    </div>
                  </label>
                  <div
                    class="chakra-input__group css-4302v8"
                  >
                    <input
                      class="chakra-input css-g75xpy"
                      data-testid="amount"
                      id="field-:r1:"
                      inputmode="numeric"
                      placeholder="Introduce amount"
                      type="text"
                      value=""
                    />
                    <div
                      class="chakra-input__right-element css-y1u7d2"
                    />
                  </div>
                </div>
              </div>
            </form>
          </div>
          <div
            class="chakra-stack css-eunu1i"
          >
            <div
              class="chakra-stack css-wqnj46"
            >
              <h2
                class="chakra-heading css-m8uvwu"
                data-testid="details-title"
              >
                Stable coin details
              </h2>
              <div
                class="css-13brihr"
              >
                <p
                  class="chakra-text css-53sllx"
                  data-testid="details-review-title"
                >
                  Basic details
                </p>
                <div
                  class="css-0"
                >
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-0"
                    >
                      <p
                        class="chakra-text css-bpirlf"
                      >
                        Name
                      </p>
                      <div
                        class="chakra-stack css-1c8ysgd"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          ManuCoin2
                        </p>
                      </div>
                    </div>
                    <hr
                      aria-orientation="horizontal"
                      class="chakra-divider css-zaaihy"
                      title="divider"
                    />
                  </div>
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-1"
                    >
                      <p
                        class="chakra-text css-bpirlf"
                      >
                        Symbol
                      </p>
                      <div
                        class="chakra-stack css-1c8ysgd"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          MCC2
                        </p>
                      </div>
                    </div>
                    <hr
                      aria-orientation="horizontal"
                      class="chakra-divider css-zaaihy"
                      title="divider"
                    />
                  </div>
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-2"
                    >
                      <p
                        class="chakra-text css-bpirlf"
                      >
                        Decimals
                      </p>
                      <div
                        class="chakra-stack css-1c8ysgd"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          3
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="css-13brihr"
              >
                <p
                  class="chakra-text css-53sllx"
                  data-testid="details-review-title"
                >
                  Optional details
                </p>
                <div
                  class="css-0"
                >
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-0"
                    >
                      <p
                        class="chakra-text css-bpirlf"
                      >
                        Initial supply
                      </p>
                      <div
                        class="chakra-stack css-1c8ysgd"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          0
                        </p>
                      </div>
                    </div>
                    <hr
                      aria-orientation="horizontal"
                      class="chakra-divider css-zaaihy"
                      title="divider"
                    />
                  </div>
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-1"
                    >
                      <p
                        class="chakra-text css-bpirlf"
                      >
                        Total supply
                      </p>
                      <div
                        class="chakra-stack css-1c8ysgd"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          100
                        </p>
                      </div>
                    </div>
                    <hr
                      aria-orientation="horizontal"
                      class="chakra-divider css-zaaihy"
                      title="divider"
                    />
                  </div>
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-2"
                    >
                      <p
                        class="chakra-text css-bpirlf"
                      >
                        Supply type
                      </p>
                      <div
                        class="chakra-stack css-1c8ysgd"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          Infinite
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-3gu2fl"
        >
          <div
            class="chakra-stack css-1amf41o"
          >
            <button
              class="chakra-button css-1nrhi9n"
              data-testid="cancel-btn"
              type="button"
            >
              Go back
            </button>
            <button
              class="chakra-button css-1178n6s"
              data-testid="confirm-btn"
              disabled=""
              type="button"
            >
              Accept
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
